src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -1797,22 +1833,33 @@ private static void ThrowOverflowOrFormatException(bool overflow, string overflo
                (Exception)new FormatException(SR.Format_InvalidString);
         }
 
-        private static bool TryNumberToDouble(ref NumberBuffer number, ref double value)
+        private static bool TryNumberToDouble(ref NumberBuffer number, out double value)
         {
-            double d = NumberToDouble(ref number);
-            if (!double.IsFinite(d))
+            ulong bits = NumberToFloatingPointBits(ref number, in FloatingPointInfo.Double);
+            double result = BitConverter.Int64BitsToDouble((long)(bits));
+
+            value = result;
+
+            if (number.Sign)
             {
-                value = default;
-                return false;
+                value = -value;
             }
 
-            if (d == 0.0)
+            return true;
+        }
+
+        private static bool TryNumberToSingle(ref NumberBuffer number, out float value)
```

Comment:
When does this method ever return something other than `true`? If it doesn't, maybe it shouldn't be a `Try` method then.

Reply:
Same for the above method.

---
In reply to: [231663340](https://github.com/dotnet/coreclr/pull/20707#discussion_r231663340) [](ancestors = 231663340)

Reply:
Will fixup. It no longer returns `false` since `Overflow` will now return the appropriately signed `Infinity`, rather than false.

Reply:
Fixed.